libpromises/evalfunction.c

```diff
@@ -8722,6 +8722,28 @@ static FnCallResult FnCallCFEngineCallers(EvalContext *ctx, ARG_UNUSED const Pol
     return FnReturnContainerNoCopy(callers);
 }
 
+static FnCallResult FnCallString(EvalContext *ctx,
+                                 const Policy *policy,
+                                 const FnCall *fp,
+                                 const Rlist *finalargs)
+{
+    // Parser doesnt allow this to be true since STRING_ARGS has one argument
+    if(!finalargs) {
+        return FnFailure();
+    }
+
+    char* ret;
+    switch(finalargs->val.type) {
+        case RVAL_TYPE_SCALAR:
+            ret = RvalToString(finalargs->val);
+            break;
+        default:
+            ProgrammingError("Shouldnt reach this TODO");
```

Comment:
You should add a case for a data container, and convert it to a json string using:

https://github.com/cfengine/libntech/blob/6092009490d4219ec160a17a70ff083b7ef3a835/libutils/json.h#L581

Reply:
Maybe this should be doing the same thing that `format("%s", anything)` does currently - no need to reimplement all that code, just factor out a stringification function.

Reply:
I got a hint from vratislav to checkout `VarNameOrInlineToJson` and i got it to work. With the current "signature" for the policy function only RVAL_TYPE_SCALAR is passed which means that a switch is unecessary.  But i still have doubts on the logics "integrity"